src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
+    // the processFulfilledOrder() method must change the OrderState to Failed by adding in that
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceFailed() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.FAILED);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
         mockCloudConnectorFactory(orderInstance);
 
-        Assert.assertNull(this.failedOrderList.getNext());
-
+        // exercise
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
-
-        Assert.assertNull(this.fulfilledOrderList.getNext());
-
+        // verify
         Order test = this.failedOrderList.getNext();
         Assert.assertNotNull(test);
         Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
         Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
+    // test case: When running thread in the FulfilledProcessor with OrderState Null must throw a
+    // ThrowableException.
+    @SuppressWarnings("static-access")
     @Test
-    public void testRunThrowableExceptionWhileTryingToProcessOrder()
+    public void testRunThrowableExceptionWhileTryingToProcessOrderStateNull()
             throws InterruptedException, UnexpectedException {
-        
+
+        // set up
         Order order = Mockito.mock(Order.class);
         OrderState state = null;
         order.setOrderState(state);
         this.fulfilledOrderList.addItem(order);
 
         spyFulfiledProcessor();
 
+        // exercise
         Mockito.doThrow(new RuntimeException()).when(this.fulfilledProcessor)
                 .processFulfilledOrder(order);
 
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
+        // verify
+        Mockito.verify(this.fulfilledProcessor).processFulfilledOrder(order);
     }
 
+    // test case: When running thread in the FulfilledProcessor with OrderState Null must throw a
+    // UnexpectedException.
+    @SuppressWarnings("static-access")
     @Test
     public void testThrowUnexpectedExceptionWhileTryingToProcessOrder()
             throws InterruptedException, UnexpectedException {
-        
+
+        // set up
         Order order = Mockito.mock(Order.class);
         OrderState state = null;
         order.setOrderState(state);
```

Comment:
I don't think `order` should be mocked. Actually, this line does nothing, since the object is a stub, right?

Reply:
In this test, only a stub is needed to set a Null OrderState, the existence of the test is related to the code coverage.